packages/rocketchat-livechat/server/lib/Livechat.js

```diff
@@ -535,7 +535,7 @@ export const Livechat = {
 		if (departmentId) {
 			let agents = Livechat.getOnlineAgents(departmentId);
 
-			if (agents.count() === 0 && settings.get('Livechat_guest_pool_with_no_agents')) {
+			if ((!agents || agents.count() === 0) && settings.get('Livechat_guest_pool_with_no_agents')) {
```

Comment:
As I could see the `getOnlineAgents` may return either an Array or a Cursor, so your test may still fail

You may need to change the returns from arrays to undefined [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-models/server/models/LivechatDepartmentAgents.js#L81) and [here](https://github.com/RocketChat/Rocket.Chat/blob/7a8e9c31d8a62821d0505d156f7517cf4e2fe8e9/packages/rocketchat-models/server/models/LivechatDepartmentAgents.js#L100)